internal/models/argo.go

```diff
@@ -168,6 +187,88 @@ func (app *Application) ListUnhealthyResources() []string {
 	return list
 }
 
+// IsManagedByWatcher checks if the application is managed by the watcher.
+// It checks if the application's metadata contains the "argo-watcher/managed" annotation with the value "true".
+func (app *Application) IsManagedByWatcher() bool {
+	if app.Metadata.Annotations == nil {
+		return false
+	}
+	return app.Metadata.Annotations[managedAnnotation] == "true"
+}
+
+func (app *Application) processAppAnnotations(annotations map[string]string, task *Task) *updater.ArgoOverrideFile {
```

Comment:
Same as with the method that's calling this, I think we should explicitly return errors on parsing the images. There's no reason to silently return nothing and then, later in the code, wait for the updated image to be deployed. "fail fast, fail often" approach is best there, I think.

We want to be as transparent as possible about what goes wrong in deployment, and as fast as possible.